functions/__fzf_search_shell_variables.fish

```diff
@@ -1,24 +1,33 @@
-function __fzf_search_shell_variables --description "Search and inspect shell variables using fzf. Insert the selected variable into the commandline at the cursor."
-    # Make sure that fzf uses fish to execute __echo_value_or_print_message, which
-    # is an autoloaded fish function so doesn't exist in other shells.
+function __fzf_search_shell_variables --argument-names variable_file --description "Search and inspect shell variables using fzf. Insert the selected variable into the commandline at the cursor."
+    if test -z "$variable_file"
+        set_color red
+        printf "\nThe signature of __fzf_search_shell_variables was changed in https://github.com/PatrickF1/fzf.fish/pull/71.\nPlease see the latest conf.d/fzf.fish and update your key bindings.\n\n"
+        set_color normal
+
+        commandline --function repaint
+        return
+    end
+
+    # Make sure that fzf uses fish to execute the preview function, which uses
+    # fish's built-in string function which doesn't exist in other shells.
     # Using --local so that it does not clobber SHELL outside of this function.
     set --local --export SHELL (command --search fish)
 
-    # Pipe the names of all shell variables to fzf and attempt to display the value
-    # of the selected variable in fzf's preview window.
-    # Non-exported variables will not be accessible to the fzf process, in which case
-    # __echo_value_or_print_message will print an informative message in lieu of the value.
+    # Exclude the history variable name from being piped into fzf because it's not included in
+    # $variable_file (the output of set --show). It's also not worth showing anyway as
+    # __fzf_search_history is a much better way to examine history.
     # We use the current token to pre-populate fzf's query. If the current token begins
     # with a $, we remove it from the query so that it will better match the variable names
     # and we put it back later when replacing the current token with the user's selection.
     set variable_name (
-        set --names |
-        fzf --preview '__fzf_display_value_or_error {}' \
+        string collect $argv[2..-1] | string match --invert history |
+        fzf --preview "string match   --regex '^\\\${}(?::|\[).+' <$variable_file |
```

Comment:
Need a bit of documentation on what these the string match and string replace commands are doing.

Reply:
Sorry I know you wanted to get rid of that extra file but I think we need a helper for the preview command so that we can more easily document the logic.

Reply:
Sure, but let's give it a name.